### PR TITLE
Issue #5947: Better prevent path alias conflicts

### LINKED ISSUE #5947: Alias malfunction when two pages have the same title in multi-language installation
## Description of the bug

When you have Language and Locale enabled and you create two pages with the same title, one with English, and then one with "Language none", the two pages end up with the same alias. This makes the second page (Language none) unreachable, unless you type the system URL (eg. `node/9`). 

## Steps To Reproduce

1. Enable Locale and Language
2. Install a new language. 
3. Enable Multilingual support for Content type > Page in `admin/structure/types/manage/page`
4. Create a page with title "Test". Be sure to select English as language, and enter "English" as text in the body field (to be able to tell that this is the English version), then save.
5. Check `admin/config/urls/path` and confirm that the `test` alias has been created, and its language is set to English.
6. Create a second page, also with title "Test", but this time select "none" for the language. Add some text like "no language" in the body field, then save.
7. After saving the new content, you are redirected to the English version of the page you created in step 4 above (the "English" text is shown in the body instead of "no language") 👎🏼 
8. If you check `admin/config/urls/path`, a second `path` alias has been created ( 👎🏼 ) with its language set to "none" ( 👍🏼 )

   image

When you visit the page by using the alias as in example.com/test, the one in English is shown. The one with "Language none" ALSO has the alias "test" (you can check that by hovering over that second page in `/admin/content`)

This makes the second page (language none) unreachable through the alias (the first is always shown). The only way to see it is by using its system url.

## Further testing
If you edit any of the pages that have the same path alias and simply save it again (no edits), then the path alias is automatically updated to `path-1` 👍🏼 

## Actual behavior

The second page (language none) has the same alias as the English one, making it unreachable through the alias.

## Expected behavior

A different alias should be created for the language none page, in this case `test-1`.

## Additional information

Backdrop 1.24.0

### PR DESCRIPTION
**Pull request by indigoxela**
Fixes https://github.com/backdrop/backdrop-issues/issues/5947

### CODE REVIEW

File: core/modules/path/path.admin.inc [hunk 1]
```diff
@@ -223,18 +250,25 @@ function path_admin_form_validate($form, &$form_state) {
...
+    ->condition('pid', $pid, '<>')
+    ->condition('alias', $alias);
+  if ($langcode != LANGUAGE_NONE) {
+    $query->condition('langcode', array($langcode, LANGUAGE_NONE), 'IN');
+  }
+  $existing = $query->execute()->fetchCol();
 
-  if ($has_alias) {
-    form_set_error('alias', t('The alias %alias is already in use in this language.', array('%alias' => $alias)));
+  if (count($existing)) {
```
Comment (herbdool) [inline core/modules/path/path.admin.inc:262]: Do we know that `$existing` will always be an array? I can't recall if `fetchCol()` will do that.

Comment (indigoxela) [reply core/modules/path/path.admin.inc:262]: According to [database.inc](https://github.com/backdrop/backdrop/blob/1.x/core/includes/database/database.inc#L2288) it's always an array.

Testing with an empty result set confirms that (empty array in that case).

Comment (herbdool) [reply core/modules/path/path.admin.inc:262]: Good

File: core/modules/path/path.admin.inc [hunk 2]
```diff
@@ -208,6 +208,33 @@ function path_admin_form_delete_submit($form, &$form_state) {
...
 }
 
+/**
+ * Get language names from codes and concatenate to string.
+ *
+ * @param array $language_codes
+ *   Array of language codes.
+ *
+ * @return string
+ *   Comma separated language names or codes, if language got removed.
```
Comment (quicksketch) [inline core/modules/path/path.admin.inc]:
```suggestion
 *   Comma separated language names (or codes, if the language name is unknown).
```

**Comment by quicksketch** [comment]
Merged into 1.x and 1.31.x. Thank you @indigoxela!
